src/include/api/ivf_pq_index.h

```diff
@@ -112,6 +113,19 @@ class IndexIVFPQ {
           id_datatype_ = string_to_datatype(value);
         } else if (key == "partitioning_index_type") {
           partitioning_index_datatype_ = string_to_datatype(value);
+        } else if (key == "distance_metric") {
+          try {
+            int metric_value = std::stoi(value);
+            if (metric_value < 0 ||
+                metric_value > static_cast<int>(DistanceMetric::COSINE)) {
+              throw std::runtime_error(
+                  "Invalid distance metric value: " + value);
+            }
+            distance_metric = static_cast<DistanceMetric>(metric_value);
+          } catch (const std::exception& e) {
+            throw std::runtime_error(
+                "Error setting distance metric: " + std::string(e.what()));
+          }
```

Comment:
Can we make this into a helper function so we don't need to write it twice?

Reply:
If you can, please take a look over what I tried to do now - I think it's pretty smooth and can be reused for future stuff

Reply:
Lgtm thanks!